UsageError in package that contains pytest_plugins module

When running `pytest --doctest-modules` on a package that contains a pytest plugin, `pytest` errors out and produces a pretty long stacktrace. The code that triggers the error is the `_get_plugin_specs_as_list` function that receives a module object instead of a list or a string.

The behavior is OS-independent as far as I can tell, but affects only pytest 3.7+ ; pytest 3.6.4 works fine. Also note that the error is *only* triggered when doc-testing the specific subpackage that contains the plugin.

For proper context, we have the following directory structure:
```
odl
 |
 |- util --
 |        |- pytest_plugins.py
 |        |- ...
 | - ...
```
The error is triggered by `pytest --doctest-modules odl/util`.

The full backtrace is [here](https://gist.github.com/kohr-h/0899f6e87262de1f16e409e46f46c1e2).
Link to the subpackage: https://github.com/odlgroup/odl/tree/master/odl/util

Edit: Updated pytest versions affected

From the traceback (please consider putting those in the issues themselves for searchability etc):
```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <_pytest.config.PytestPluginManager object at 0x0000000002B84240>
mod = <module 'odl.util' from '[...]\\odl\\util\\__init__.py'>

    def consider_module(self, mod):
>       self._import_plugin_specs(getattr(mod, "pytest_plugins", []))

[...]\_pytest\config\__init__.py:462:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <_pytest.config.PytestPluginManager object at 0x0000000002B84240>
spec = <module 'odl.util.pytest_plugins' from '[...]\\odl\\util\\pytest_plugins.py'>

    def _import_plugin_specs(self, spec):
>       plugins = _get_plugin_specs_as_list(spec)

[...]\_pytest\config\__init__.py:465:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

specs = <module 'odl.util.pytest_plugins' from '[...]\\odl\\util\\pytest_plugins.py'>

    def _get_plugin_specs_as_list(specs):
        """
        Parses a list of "plugin specs" and returns a list of plugin names.

        Plugin specs can be given as a list of strings separated by "," or already as a list/tuple in
        which case it is returned as a list. Specs can also be `None` in which case an
        empty list is returned.
        """
        if specs is not None:
            if isinstance(specs, str):
                specs = specs.split(",") if specs else []
            if not isinstance(specs, (list, tuple)):
                raise UsageError(
                    "Plugin specs must be a ','-separated string or a "
>                   "list/tuple of strings for plugin names. Given: %r" % specs
                )
E               _pytest.config.exceptions.UsageError: Plugin specs must be a ','-separated string or a list/tuple of strings for plugin names. Given: <module 'odl.util.pytest_plugins' from '[...]\\odl\\util\\pytest_plugins.py'>

[...]\_pytest\config\__init__.py:518: UsageError
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! KeyboardInterrupt !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
[...]\_pytest\_code\source.py:341: KeyboardInterrupt
(to show a full traceback on KeyboardInterrupt use --fulltrace)
========================== 12 error in 5.32 seconds ===========================
```

It looks like `odl.util.pytest_plugins` is expected to be a list, but is a
module:

```

self = <_pytest.config.PytestPluginManager object at 0x0000000002B84240>
mod = <module 'odl.util' from '[...]\\odl\\util\\__init__.py'>

    def consider_module(self, mod):
>       self._import_plugin_specs(getattr(mod, "pytest_plugins", []))

[...]\_pytest\config\__init__.py:462:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <_pytest.config.PytestPluginManager object at 0x0000000002B84240>
spec = <module 'odl.util.pytest_plugins' from '[...]\\odl\\util\\pytest_plugins.py'>

    def _import_plugin_specs(self, spec):
>       plugins = _get_plugin_specs_as_list(spec)
```

I don't know the internals really, but it looks like pytest should skip / not consider this?!

--- a/changelog/3899.bugfix.rst
+++ b/changelog/3899.bugfix.rst
@@ -1,0 +1,1 @@
+Do not raise ``UsageError`` when an imported package has a ``pytest_plugins.py`` child module.

--- a/src/_pytest/config/__init__.py
+++ b/src/_pytest/config/__init__.py
@@ -559,8 +559,8 @@ def _get_plugin_specs_as_list(specs):
     which case it is returned as a list. Specs can also be `None` in which case an
     empty list is returned.
     """
-    if specs is not None:
-        if isinstance(specs, str):
+    if specs is not None and not isinstance(specs, types.ModuleType):
+        if isinstance(specs, six.string_types):
             specs = specs.split(",") if specs else []
         if not isinstance(specs, (list, tuple)):
             raise UsageError(

--- a/testing/acceptance_test.py
+++ b/testing/acceptance_test.py
@@ -969,6 +969,20 @@ def test_import_plugin_unicode_name(testdir):
     assert r.ret == 0
 
 
+def test_pytest_plugins_as_module(testdir):
+    """Do not raise an error if pytest_plugins attribute is a module (#3899)"""
+    testdir.makepyfile(
+        **{
+            "__init__.py": "",
+            "pytest_plugins.py": "",
+            "conftest.py": "from . import pytest_plugins",
+            "test_foo.py": "def test(): pass",
+        }
+    )
+    result = testdir.runpytest()
+    result.stdout.fnmatch_lines("* 1 passed in *")
+
+
 def test_deferred_hook_checking(testdir):
     """
     Check hooks as late as possible (#1821).

